plugins/modules/rds_cluster_param_group.py

```diff
@@ -0,0 +1,369 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: Contributors to the Ansible project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r"""
+---
+module: rds_cluster_param_group
+version_added: 7.5.0
+short_description: Manage RDS cluster parameter groups
+description:
+  - Creates, modifies, and deletes RDS cluster parameter groups.
+options:
+  state:
+    description:
+      - Specifies whether the RDS cluster parameter group should be present or absent.
+    default: present
+    choices: [ 'present' , 'absent' ]
+    type: str
+  name:
+    description:
+      - The name of the RDS cluster parameter group to create, modify or delete.
+    required: true
+    type: str
+  description:
+    description:
+      - The description for the RDS cluster parameter group.
+      - Required for O(state=present).
+    type: str
+  db_parameter_group_family:
+    description:
+      - The RDS cluster parameter group family name.
+      - An RDS cluster parameter group can be associated with one and only one RDS cluster parameter group family,
+        and can be applied only to a RDS cluster running a database engine and engine version compatible with that RDS cluster parameter group family.
+      - Please use M(amazon.aws.rds_engine_versions_info) module To list all of the available parameter group families for a DB engine.
+      - The RDS cluster parameter group family is immutable and can't be changed when updating a RDS cluster parameter group.
+        See U(https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbclusterparametergroup.html)
+      - Required for O(state=present).
+    type: str
+  immediate:
+    description:
+      - Whether to apply the changes immediately, or after the next reboot of any associated instances.
+      - Ignored when O(state=absent)
+    aliases:
+      - apply_immediately
+    type: bool
+  params:
+    description:
+      - Map of parameter names and values. Numeric values may be represented as K for kilo (1024), M for mega (1024^2), G for giga (1024^3),
+        or T for tera (1024^4), and these values will be expanded into the appropriate number before being set in the parameter group.
+    aliases: [parameters]
+    type: dict
```

Comment:
Is there a specific reason for not utilizing a list instead of a dictionary here? Based on [this](https://boto3.amazonaws.com/v1/documentation/api/1.26.93/reference/services/rds/client/modify_db_parameter_group.html), it appears that parameters are accepted as a list of dictionaries.

Reply:
updated